nifi/nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/main/java/org/apache/nifi/processors/kafka/PutKafka.java

```diff
@@ -136,6 +136,68 @@
             .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
             .expressionLanguageSupported(false)
             .build();
+    public static final PropertyDescriptor PRODUCER_TYPE = new PropertyDescriptor.Builder()
+            .name("Producer Type")
+            .description("This parameter specifies whether the messages are sent asynchronously in a background thread."
+                    + " Valid values are (1) async for asynchronous send and (2) sync for synchronous send."
+                    + " By setting the producer to async we allow batching together of requests (which is great for throughput)"
+                    + " but open the possibility of a failure of the client machine dropping unsent data.")
+            .required(true)
+            .allowableValues("sync", "async")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(false)
+            .defaultValue("sync")
+            .build();
+        public static final PropertyDescriptor BATCH_NUM_MESSAGES = new PropertyDescriptor.Builder()
+            .name("Async Message Batch Size (batch.num.messages)")
```

Comment:
I would avoid adding the Kafka property names in parens. This results in very long property names that are likely to be cutoff in the UI unless the user expands the column. They are also relevant only for those who are used to writing Kafka clients, so that the understand the meaning of those property names. For those who don't write clients (especially for non-developers) I think it makes the property names confusing.